agent-control/src/cli/on_host/proxy_config.rs

```diff
@@ -0,0 +1,120 @@
+//!  This module defines the proxy configuration for on-hots commands.
+
+use std::convert::Infallible;
+
+use serde::Serialize;
+
+/// Holds the proxy configuration.
+/// Cannot use [crate::http::config::ProxyConfig] directly due lack of support for defaults in clap.
+/// See <https://github.com/clap-rs/clap/issues/4746> for details.
+#[derive(Debug, Default, Clone, PartialEq, Serialize, clap::Args)]
+pub struct ProxyConfig {
+    #[serde(skip_serializing_if = "is_none_or_empty_string", rename = "url")]
+    #[arg(long, required = false)]
+    pub proxy_url: Option<String>,
+
+    #[serde(
+        skip_serializing_if = "is_none_or_empty_string",
+        rename = "ca_bundle_dir"
+    )]
+    #[arg(long, required = false)]
+    pub proxy_ca_bundle_dir: Option<String>,
+
+    #[serde(
+        skip_serializing_if = "is_none_or_empty_string",
+        rename = "ca_bundle_file"
+    )]
+    #[arg(long, required = false)]
+    pub proxy_ca_bundle_file: Option<String>,
+
+    #[serde(skip_serializing_if = "is_false")]
+    #[arg(long, default_value_t = false, value_parser = ignore_system_proxy_parser, action = clap::ArgAction::Set)]
+    pub ignore_system_proxy: bool,
+}
+
+impl ProxyConfig {
+    pub fn is_empty(&self) -> bool {
```

Comment:
nit: would be possible to skip serializing the whole proxy config in case it's equal to the default value, would make sense? perhaps it cannot be done because is flattened

Reply:
I tried to, but I didn't find a way to do it because of the flattened thing 🙁